### ISSUE TITLE
feat(embeddings): Qwen3-VL-Embedding and Llama-Nemotron-VL-Embed (text and image inputs, one vector per input)

### ISSUE THREAD

**Issue by inureyes** [lablup/mlxcel#1345]
## Summary

Two multimodal embedders reuse VLM stacks mlxcel already runs. Qwen3-VL-Embedding (`model_type: qwen3_vl`, `architectures: ["Qwen3VLForConditionalGeneration"]`, `1_Pooling` `pooling_mode: lasttoken`) wraps its input in the checkpoint's chat template with an instruction system message, runs the causal Qwen3-VL stack (vision tower, DeepStack injection, M-RoPE) and pools the last real token. Llama-Nemotron-VL-Embed (`model_type: llama_nemotron_vl`, `architectures: ["LlamaNemotronVLModel"]`) composes a SigLIP-400M tower, pixel-shuffle plus an `mlp1` projector and a bidirectional Llama 3.2 1B, then mean-pools. This issue adds both as `EmbeddingModel` implementations with `supports_images() == true`, so `/v1/embeddings` accepts `{type: image_url}` items for them.

## Current behavior

- `src/models/detection.rs:320` routes `qwen3_vl` to the generative `ModelType::Qwen3VL`; `crate::models::Qwen3VLModel::forward_impl` (`src/models/qwen3_vl.rs:829`) always applies `lm_head` (non-optional field at line 594) and builds causal masks internally.
- `llama_nemotron_vl` has no detection arm. The pieces exist separately: `src/vision/encoders/siglip.rs:484` (`SigLipVisionModel`, post-layernorm output), `src/vision/internvl.rs:56-118` (`mlp1` connector and `pixel_shuffle(downsample_ratio)`), `src/vision/processors/internvl.rs:45-60` (`InternVLProcessor` dynamic tiling with ImageNet mean/std) and `Llama3Model`.
- #1353 accepts image items only for models with `supports_images() == true`; none exist yet.

### Qwen3-VL-Embedding

Config (verified on `Qwen/Qwen3-VL-Embedding-2B`): text `hidden_size 2048`, 28 layers, `tie_word_embeddings true`, `rope_theta 5e6` with the Qwen3-VL M-RoPE sections; vision `hidden_size 1024`, `patch_size 16`, DeepStack indexes as in the generative 2B; `image_token_id 151655`; `1_Pooling/config.json` `{pooling_mode: "lasttoken", include_prompt: true}`; `sentence_bert_config.json` `processing_kwargs.chat_template.add_generation_prompt: true`; `config.normalize` absent (default true); `max_length = 8192`.

Text formatting (`EmbeddingModel::format_text`): the checkpoint's `chat_template.jinja` applied to
```
[{"role": "system", "content": [{"type": "text", "text": <instruction or "Represent the user's input.">}]},
 {"role": "user",   "content": [ {"type": "image"} (when an image is given), {"type": "text", "text": <text>} ]}]
```
with `add_generation_prompt = true`. An instruction that does not end in punctuation gets a trailing `.`. The rendered prompt is `<|im_start|>system\n{instruction}<|im_end|>\n<|im_start|>user\n[<|vision_start|><|image_pad|><|vision_end|>]{text}<|im_end|>\n<|im_start|>assistant\n`, and the pooled token is the final `\n` of the assistant header. The existing `ChatTemplateProcessor` (`src/server/chat_template.rs`) renders it; image items use the same processor path as generative Qwen3-VL (`src/vision/processors/qwen2_vl.rs`) to expand `<|image_pad|>` to the patch count.

Forward:
```
embeds, grid = Qwen3VLModel::get_input_embeddings(input_ids, pixel_values, grid_thw)  # or embed_tokens for text
mask = create_causal_padding_mask(attention_mask, 0)           # [B,1,L,L]; right padding
h = text_model.forward_hidden(input_ids, Some(embeds), fresh caches, Some(mask))  # M-RoPE positions + DeepStack, no lm_head
embedding = normalize_l2(lasttoken_pool(h, attention_mask))   # normalize unless config.normalize == false
```
A batch of text-only inputs is right-padded and embedded together; image inputs are embedded one at a time (M-RoPE deltas are per sequence).

Weights: the generative Qwen3-VL layout, loaded by the existing VLM loader (`src/loading/vlm_qwen.rs`); `lm_head` is tied and unused.

### Llama-Nemotron-VL-Embed

Config (verified on `nvidia/llama-nemotron-embed-vl-1b-v2`): top level `downsample_ratio 0.5`, `img_context_token_id 128258` (`<IMG_CONTEXT>`), `force_image_size 512`, `select_layer -1`, `pooling "avg"`; `text_config` Llama (`hidden_size 2048`, 16 layers, 32/8 heads, `vocab_size 128266`, llama3 `rope_scaling`); `vision_config` `siglip_vision_model` (`hidden_size 1152`, 27 layers, `intermediate_size 4304`, `image_size 512`, `patch_size 16`, 1024 position rows); `1_Pooling` mean; `processor_config.json`: `image_size 512`, `max_input_tiles 6`, `use_thumbnail true`, `norm_type "siglip"` (mean and std `0.5`), `num_image_token 256`, `q_max_length 512`, `p_max_length 4096`, `query_prefix "query:"`, `passage_prefix "passage:"`.

Prompt construction (caller supplies the `query: ` / `passage: ` prefix as for the text-only sibling; the server supplies the image tokens): for an image item the text becomes `passage: <image> {text}` where `<image>` expands to `<img>` + `<IMG_CONTEXT>` repeated `256 * num_tiles` + `</img>`; `num_tiles` comes from the InternVL dynamic tiling at 512 px (1 to 6 tiles plus a thumbnail when more than one). Tokenization adds `<|begin_of_text|>` (the tokenizer's post-processor); `pad_token` is `<|finetune_right_pad_id|>`; `max_length = min(4096, 8192)`.

Forward:
```
h = embed_tokens[input_ids]                                             # [B, L, 2048]
if pixel_values:                                                        # [tiles, 512, 512, 3] channels-last
    v = SigLipVisionModel.forward(pixel_values).hidden_states            # post_layernorm output, [tiles, 1024, 1152]; the attention-pooling head is not used
    v = pixel_shuffle(v, 0.5)                                            # [tiles, 256, 4608]
    v = mlp1.3(gelu(mlp1.1(LayerNorm_mlp1.0(v))))                       # 4608 -> 2048 -> 2048
    h[input_ids == 128258] = v.reshape(-1, 2048)                         # merge in order (merge::merge_llava)
mask = create_bidirectional_padding_mask(attention_mask)                # [B,1,1,L]
for layer: h = layer.forward(h, fresh KVCache, Some(mask))               # llama3::TransformerBlock::forward
h = norm(h); embedding = normalize_l2(mean_pool(h, attention_mask))
```

Weight keys (verified):
```
language_model.embed_tokens.weight [128266, 2048], language_model.layers.{i}.*, language_model.norm.weight
vision_model.vision_model.embeddings.patch_embedding.{weight [1152,3,16,16], bias}
vision_model.vision_model.embeddings.position_embedding.weight [1024, 1152]
vision_model.vision_model.encoder.layers.{i}.{layer_norm1,layer_norm2,self_attn.{q,k,v,out}_proj,mlp.{fc1,fc2}}.{weight,bias}
vision_model.vision_model.post_layernorm.{weight,bias}
vision_model.vision_model.head.*            (attention-pooling head: probe, attention.in_proj_{weight,bias}, attention.out_proj, layernorm, mlp) -> drop
mlp1.0.{weight,bias} [4608], mlp1.1.{weight [2048,4608],bias}, mlp1.3.{weight [2048,2048],bias}
```
Sanitize: drop `vision_model.vision_model.head.*`, `rotary_emb.inv_freq`, `lm_head.*`, `position_ids`; transpose `patch_embedding.weight` to channels-last `[1152, 16, 16, 3]` as `src/loading/vlm_siglip.rs` does; map `language_model.` to `language_model.model.` when the inner `model.` is missing (the `Llama3Model` loader expects `model.`).

## Implementation plan

1. `src/models/qwen3_vl.rs`: add `pub(crate) fn forward_hidden(&self, input_ids, input_embeddings, caches, mask) -> [B, L, D]` (the body of `forward_for_sequence` up to `norm`, keeping M-RoPE position handling and DeepStack injection), and make `forward_for_sequence` call it then `lm_head`. `// Used by: Qwen3VL generation, Qwen3VLEmbedding, #1356 (Qwen3-VL reranker)`.
2. `src/models/qwen3_vl_embedding.rs` (+ `_tests.rs`): `Qwen3VLEmbeddingModel { vlm: crate::vision::Qwen3VLModel, chat_template: ChatTemplateProcessor, pooling, normalize, default_instruction }`; `load(dir)` reuses the Qwen3-VL VLM loader; `impl EmbeddingModel` (`supports_images = true`, `format_text` renders the messages above, `embed` as above, `default_pooling = LastToken`, `embedding_dim = text hidden_size`).
3. `src/models/llama_nemotron_vl_embedding.rs` (+ `_tests.rs`): `LlamaNemotronVLEmbeddingModel { vision: SigLipVisionModel, connector: InternVLConnector (reuse `src/vision/internvl.rs` mlp1 + pixel_shuffle with prefix `mlp1`), text: (embed_tokens, layers, norm) via `Llama3Backbone::from_weights_without_head` from #1325 or a local equivalent, processor: InternVLProcessor::new(512, 1, 6, true) with mean/std 0.5, img_context_token_id, num_image_token: 256, pooling, normalize }`; `format_text` inserts the `<img>`/`<IMG_CONTEXT>`/`</img>` block when an image is attached; `embed` as above.
4. `src/models/detection.rs`: in `is_embedding_checkpoint`, `qwen3_vl` with a `modules.json` Pooling module -> `ModelType::Qwen3VLEmbedding`; `llama_nemotron_vl` -> `ModelType::LlamaNemotronVLEmbedding`. `src/model_metadata.rs` entries; `src/embeddings/loader.rs` arms.
5. `src/server/routes/embeddings.rs`: no schema change; image items already route to `embed_image` for `supports_images()` models (#1353). Add `instruction` pass-through to `format_text`.
6. `docs/supported-models.md` rows in the multimodal embedding table; `docs/embeddings.md`: the instruction default for Qwen3-VL-Embedding and the `query:`/`passage:` prefixes plus tile budget for Llama-Nemotron-VL-Embed.

## Validation

(a) Unit tests:
- `src/models/qwen3_vl_embedding_tests.rs`: `format_text_renders_system_instruction_and_generation_prompt` (exact string above), `instruction_gets_trailing_period`, `forward_hidden_then_head_matches_forward` (random 2-layer text config, token-exact), `last_token_pool_ignores_right_padding`.
- `src/models/llama_nemotron_vl_embedding_tests.rs`: `image_block_expands_to_256_tokens_per_tile` (`<img>` + 256 x `<IMG_CONTEXT>` + `</img>` for one tile, 7 x 256 for six tiles plus thumbnail), `sanitize_drops_vision_head_and_maps_language_model_prefix`, `pixel_shuffle_then_mlp1_shapes` (`[1, 1024, 1152]` -> `[1, 256, 2048]` with random weights), `bidirectional_prefill_sees_future_tokens` (96 tokens).

(b) Real checkpoints:
```bash
./target/release/mlxcel download Qwen/Qwen3-VL-Embedding-2B
./target/release/mlxcel embed -m models/Qwen3-VL-Embedding-2B -p "A woman playing with her dog on a beach" -p "A person and a pet by the sea" -p "Quarterly revenue report for a software company"
./target/release/mlxcel-server -m models/Qwen3-VL-Embedding-2B --port 8080
curl -s localhost:8080/v1/embeddings -H 'Content-Type: application/json' -d '{
  "input": [{"type": "image_url", "image_url": {"url": "data:image/jpeg;base64,<a beach photo>"}},
            {"type": "text", "text": "A woman playing with her dog on a beach"},
            {"type": "text", "text": "Quarterly revenue report for a software company"}]}'
```
Acceptance: 2048-dim unit vectors; text-text: the paraphrase beats the unrelated sentence by at least 0.15; image-text: the beach caption beats the revenue sentence for the beach photo by at least 0.1; a text-only batch of three matches the three single-input vectors within 1e-3; no NaN in f16 (the 2B runs bf16 -> f16 by the text rule).

```bash
./target/release/mlxcel download nvidia/llama-nemotron-embed-vl-1b-v2
./target/release/mlxcel embed -m models/llama-nemotron-embed-vl-1b-v2 -p "query: what does the chart say about 2023 revenue" -p "passage: Revenue grew 12% in 2023 driven by subscriptions." -p "passage: The museum opens at nine in the morning."
curl -s localhost:8080/v1/embeddings -H 'Content-Type: application/json' -d '{
  "input": [{"type": "text", "text": "query: what does the chart say about 2023 revenue"},
            {"type": "image_url", "image_url": {"url": "data:image/png;base64,<a revenue bar chart>"}}]}'
```
Acceptance: 2048-dim unit vectors; the revenue passage beats the museum passage by at least 0.15; the chart image scores above the museum passage for the revenue query; a 6-tile landscape image produces exactly `7 * 256` image tokens and embeds without error; a padded mixed-length text batch matches single-input results within 1e-3.

## Acceptance criteria

- [x] `Qwen3VLEmbeddingModel` renders the instruction chat template, runs the causal Qwen3-VL stack without `lm_head` through `forward_hidden`, and pools the last real token for text and image inputs
- [x] `LlamaNemotronVLEmbeddingModel` composes SigLIP + pixel-shuffle + `mlp1` + bidirectional Llama with mean pooling and the `<img>`/`<IMG_CONTEXT>`/`</img>` expansion
- [x] both checkpoints pass the gates above through `mlxcel embed` and `POST /v1/embeddings`, including image items
- [x] docs/supported-models.md updated
- [x] detection table in src/models/detection.rs updated with a test
- [x] test suite passes (validated on the CUDA host as `cargo test --profile test-fast --features cuda --lib -- --test-threads=1 models::qwen3_vl models::llama_nemotron_vl` and `... embeddings::`; the metal/accelerate combination has no host here)
- [x] `cargo clippy --profile test-fast --features cuda --lib --bins --tests -- -D warnings` and `cargo fmt --all -- --check` pass

## Out of scope

- Video inputs for Qwen3-VL-Embedding.
- Qwen3-VL-Embedding-8B (same code).
- Vision feature caching across requests.

## Dependencies

Blocked by #1353

- Part of epic #1348
